### ISSUE TITLE
How to use example iOS12

### ISSUE THREAD

**Issue by chihiro-yabuta** [flutter/flutter#142103]
### Steps to reproduce

I already posted issue #142022, and reporter mentioned #94491 as duplicated issue.
I couldn't understand this thread perfectly, but I understood that this factor is the M1 mac silicon chip.
I tried to use and modify iOS12 example like below:
1. modify API KEY part
2. check PRODUCT_BUNDLE_IDENTIFIER

but it doesn't work.
how should I use this example or which example should I use in M1 mac silicon chip?

### Expected results

I expect to show google map.

### Actual results

google map is not showing.

### Code sample

I used this [example](https://github.com/flutter/packages/tree/main/packages/google_maps_flutter/google_maps_flutter_ios/example/ios12).

### Screenshots or Video

please check #142022

### Logs

please check #142022

### Flutter Doctor output

please check #142022

**Comment by huycozy** [comment]
@chihiro-yabuta Could you try the workaround at https://github.com/flutter/flutter/issues/94491#issuecomment-1488713501?

**Comment by chihiro-yabuta** [comment]
@huycozy 
I tried to copy and modify:
- ios 12 -> 13
- comment out arm line

but it also doesn't work.
[this is that repo.](https://github.com/chihiro-yabuta/google_maps_flutter_example)

**Comment by huycozy** [comment]
I asked my colleague to check this on simulator on her M1 machine. Strange thing that your both examples (in this issue and on https://github.com/flutter/flutter/issues/142022) can render map normally. She checked this on iOS 17.2 simulator. Could you try this on another simulator?

**Comment by chihiro-yabuta** [comment]
@huycozy 
I'll try to check on online simulator. thank you.

**Comment by huycozy** [comment]
I'm not sure if you accidentally closed this or if you already solved the issue. You can try checking if you have installed Rosetta 2 or not:

```console
/usr/bin/pgrep -q oahd && echo Yes || echo No
```

Please also follow the [Software requirements guideline](https://docs.flutter.dev/get-started/install/macos/mobile-android?tab=vscode#software-requirements) if you haven't done it yet.

**Comment by chihiro-yabuta** [comment]
I think my mac is satisfied with that requirements.

/usr/bin/pgrep -q oahd && echo Yes || echo No
Yes

flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.8, on macOS 13.6 22G120 darwin-arm64, locale
    en-JP)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] IntelliJ IDEA Community Edition (version 2023.2.2)
[✓] VS Code (version 1.85.2)
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!

**Comment by huycozy** [comment]
>Google Maps SDK for iOS cannot connect or validate APIKey

I'm unsure which project you are running on, but please ensure your API key is still valid with the bundle ID. You can check [this](https://stackoverflow.com/questions/19327933/google-maps-sdk-for-ios-cannot-connect-or-validate-apikey) as well.

**Comment by chihiro-yabuta** [comment]
> Google Maps SDK for iOS cannot connect or validate APIKey

this error happened both of project.
is it necessary to set bundle ID?
actually I set no limit about my api key and you could get google map.

**Comment by huycozy** [comment]
>is it necessary to set bundle ID?
actually I set no limit about my api key and you could get google map.

Does this mean the map only can't load if you restrict API key to a specified iOS app with bundle ID? 

I just found [this](https://stackoverflow.com/a/15627865/5921933) which mentioned `Case Sensitive` in the bundle ID could cause the error above. Can you try changing bundle ID to all lowercase to see if it works?

Also, just to confirm, did you enable `Maps SDK for iOS`?

**Comment by chihiro-yabuta** [comment]
@huycozy 
I already mentioned at #142022 , I set Maps SDK for iOS correctly.
and I cannot load map regardless of any restriction.
but you (it means @huycozy and your co-worker) succeeded to load map to use my api key right?
my api key doesn't restrict anything so I guess it's unnecessary to set bundle ID.

**Comment by huycozy** [comment]
>but you (it means @huycozy and your co-worker) succeeded to load map to use my api key right?

Yes, map can load well on my Intel machine and my co-worker with her M1 machine. 

>I already mentioned at https://github.com/flutter/flutter/issues/142022 , I set Maps SDK for iOS correctly.

This is what I meant. It isn't mentioned so I just want to confirm this. You can search `Maps SDK for iOS` in https://console.cloud.google.com and enable it.

![Screenshot 2024-01-29 at 19 18 15](https://github.com/flutter/flutter/assets/104349824/d259e92f-7bbc-4215-82d0-e6a1ffd12f1c)

**Comment by chihiro-yabuta** [comment]
@huycozy 
![Screenshot 2024-01-30 at 13 12 33](https://github.com/flutter/flutter/assets/114799344/45dc6c25-e1dc-4727-bfdb-d5e31613e47f)
it's enable.
